lib/pages/chat/chat_app_bar_title.dart

```diff
@@ -94,71 +95,120 @@ class ChatAppBarTitle extends StatelessWidget {
                   overflow: TextOverflow.ellipsis,
                   style: ChatAppBarTitleStyle.appBarTitleStyle(context),
                 ),
-                _buildStatusContent(context, room!),
+                _ChatAppBarStatusContent(
+                  getStreamInstance: getStreamInstance,
+                  context: context,
+                  room: room!,
+                ),
               ],
             ),
           ),
         ],
       ),
     );
   }
+}
 
-  StreamBuilder<ConnectivityResult> _buildStatusContent(
-    BuildContext context,
-    Room room,
-  ) {
-    final TextStyle? statusTextStyle =
-        ChatAppBarTitleStyle.statusTextStyle(context);
+class _ChatAppBarStatusContent extends StatelessWidget {
+  const _ChatAppBarStatusContent({
+    required this.getStreamInstance,
+    required this.context,
+    required this.room,
+  });
+
+  final Stream<ConnectivityResult> getStreamInstance;
+  final BuildContext context;
+  final Room room;
 
-    return StreamBuilder<ConnectivityResult>(
-      stream: getStreamInstance,
+  @override
+  Widget build(BuildContext context) {
+    final connectivityResultStream = getStreamInstance;
+    final directChatPresenceStream = room.directChatPresenceStream;
+
+    return StreamBuilder<List>(
+      stream: StreamZip([connectivityResultStream, directChatPresenceStream])
+          .asBroadcastStream(),
```

Comment:
why we need it as `broadcast stream`?

Reply:
hope to have a demo for StreamZip.

Reply:
```
    test('StreamZip example', () {
      final stream1 = Stream<int>.fromIterable([1, 2, 3, 4, 5]);
      final stream2 = Stream<int>.fromIterable([10, 20, 30, 40, 50]);
      StreamZip([stream1, stream2]).listen((event) {
        print(event);
      });
    });
```

```
[1, 10]
[2, 20]
[3, 30]
```

Reply:
because I need a stream object to use streambuild and listen() function returns a streamsubscription

Reply:
why not stream merge?

Reply:
but stream zip only zip events of stream which have equal length.

Reply:
@drminh2807 suggested me to use combineLatest (https://pub.dev/documentation/rxdart/latest/rx/CombineLatestStream-class.html) instead. @hoangdat please check again :D